gplugins/design_recipe/ConstituentRecipes.py

```diff
@@ -28,7 +25,7 @@ class ConstituentRecipes:
     # useful when the iterator is accessed by multiple threads.
     iter_lock: threading.Lock
 
-    def __init__(self, recipes: List[dr.DesignRecipe] = []):
+    def __init__(self, recipes: list[dr.DesignRecipe]):
```

Comment:
a design recipe should always have a non-`None` constituent recipe, right? is the idea to just pass an explicit empty list for that case (the one where a DR has no dependencies)?